src/Uno.UI/UI/Xaml/UIElement.Layout.netstd.cs

```diff
@@ -204,46 +206,42 @@ private void DoMeasure(Size availableSize)
 					break;
 				}
 
-				if (IsMeasureDirtyPath)
-				{
-					ClearLayoutFlags(LayoutFlag.MeasureDirtyPath);
+				// isMeasureDirtyPath is always true here
+				ClearLayoutFlags(LayoutFlag.MeasureDirtyPath);
 
-					// The dirty flag is set on one of the descendents:
-					// it will bypass the current element's MeasureOverride()
-					// since it shouldn't produce a different result and it's
-					// just a waste of precious CPU time to call it.
-					var children = GetChildren().GetEnumerator();
+				// The dirty flag is set on one of the descendents:
+				// it will bypass the current element's MeasureOverride()
+				// since it shouldn't produce a different result and it's
+				// just a waste of precious CPU time to call it.
+				var children = GetChildren().GetEnumerator();
 
-					//foreach (var child in children)
-					while(children.MoveNext())
+				//foreach (var child in children)
+				while(children.MoveNext())
+				{
+					if (children.Current is { IsMeasureOrMeasureDirtyPath: true } child)
 					{
-						var child = children.Current;
 						// If the child is dirty (or is a path to a dirty descendant child),
 						// We're remeasuring it.
 
-						if (child.IsMeasureOrMeasureDirtyPath)
+						var previousDesiredSize = child.DesiredSize;
+						child.Measure(child.LastAvailableSize);
+						if (child.DesiredSize != previousDesiredSize)
 						{
-							var previousDesiredSize = child.DesiredSize;
-							child.Measure(child.LastAvailableSize);
-							if (child.DesiredSize != previousDesiredSize)
-							{
-								isDirty = true;
-								break;
-							}
+							isDirty = true;
+							break;
 						}
 					}
+				}
 
-					children.Dispose(); // no "using" operator here to prevent an implicit try-catch on Wasm
+				children.Dispose(); // no "using" operator here to prevent an implicit try-catch on Wasm
 
-					if (!isDirty)
-					{
-						break;
-					}
-				}
-				else
+				if (isDirty)
 				{
-					break;
+					Console.WriteLine($"{this}: Looping for measure, {remainingTries} tries remaining...");
```

Comment:
left over debug msg? maybe convert this to `_log` call or just remove?

Reply:
if you choose to remove this, you can also simplify `if (condition) continue; break;` down to `if (!condition) break;`